diff --git a/containers/controller/config/api/Dockerfile b/containers/controller/config/api/Dockerfile
index 4d66a39..05006d7 100644
--- a/containers/controller/config/api/Dockerfile.centos
+++ b/containers/controller/config/api/Dockerfile.centos
@@ -7,6 +7,8 @@ RUN yum install -y python-neutron-lib && \
     yum clean all -y && \
     rm -rf /var/cache/yum
 
+RUN yum install -y python-pip && pip install amqp==1.4.7
+
 COPY entrypoint.sh /
 
 ENTRYPOINT ["/entrypoint.sh"]
